Modules/atexitmodule.c

```diff
@@ -133,8 +151,12 @@ atexit_callfuncs(struct atexit_state *state)
         }
 
         // bpo-46025: Increment the refcount of cb->func as the call itself may unregister it
-        PyObject* the_func = Py_NewRef(cb->func);
-        PyObject *res = PyObject_Call(cb->func, cb->args, cb->kwargs);
+        PyObject *the_func = Py_NewRef(cb->func);
+        PyObject *the_args = cb->args;
+        PyObject *the_kwargs = cb->kwargs;
+        // Unlock for re-entrancy problems
+        _PyAtExit_UNLOCK(state);
```

Comment:
Maybe check that the lock has been held here (or does UNLOCK takes care of it?)

Reply:
`PyMutex_Unlock` already does that (it throws a fatal error if you try to unlock an unlocked mutex).

Reply:
Ah I wasn't sure about it and couldn't check it easily.Thanks!

Reply:
Enjoy your travels!